debezium-core/src/main/java/io/debezium/schema/SchemaBuilderFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.schema;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+import io.debezium.connector.AbstractSourceInfo;
+import io.debezium.heartbeat.HeartbeatImpl;
+import io.debezium.pipeline.txmetadata.TransactionMonitor;
+import io.debezium.util.SchemaNameAdjuster;
+
+/**
+ * A factory for creating {@link SchemaBuilder} structs.
+ *
+ * @author Anisha Mohanty
+ */
+
+public class SchemaBuilderFactory {
+
+    private static final int SCHEMA_VERSION = 1;
+
+    private static final String SERVER_NAME_CLASS = "io.debezium.connector.common.ServerNameKey";
+    private static final String HEARTBEAT_CLASS = "io.debezium.connector.common.Heartbeat";
+
+    private static final String TRANSACTION_BLOCK_NAME = "event.block";
+    private static final String TRANSACTION_EVENT_COUNT_COLLECTION_NAME = "event.collection";
+    private static final String TRANSACTION_METADATA_KEY_CLASS = "io.debezium.connector.common.TransactionMetadataKey";
+    private static final String TRANSACTION_METADATA_VALUE_CLASS = "io.debezium.connector.common.TransactionMetadataValue";
+
+    public static Schema heartbeatKeySchema(SchemaNameAdjuster adjuster) {
```

Comment:
Please don't use static methods but instance methods and convert the class into a singleton.
That way we could easily intorduce subclasses and use inheritance.